src/examples/example_event_error_details.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::mpsc::channel;
+
+use clap::{load_yaml, App};
+use codec::Decode;
+use keyring::AccountKeyring;
+use sp_core::crypto::Pair;
+use sp_runtime::AccountId32 as AccountId;
+use sp_runtime::MultiAddress;
+
+use substrate_api_client::rpc::WsRpcClient;
+use substrate_api_client::sp_runtime::app_crypto::sp_core::sr25519;
+use substrate_api_client::{Api, ApiResult, XtStatus};
+
+// Look at the how the transfer event looks like in in the metadata
+#[derive(Decode)]
+struct TransferEventArgs {
+    from: AccountId,
+    to: AccountId,
+    value: u128,
+}
+
+fn main() {
+    env_logger::init();
+    let url = get_node_url_from_cli();
+
+    // initialize api and set the signer (sender) that is used to sign the extrinsics
+    let from = AccountKeyring::Alice.pair();
+
+    let client = WsRpcClient::new(&url);
+    let api = Api::<sr25519::Pair, _>::new(client)
+        .map(|api| api.set_signer(from.clone()))
+        .unwrap();
+
+    let from_account_id = AccountKeyring::Alice.to_account_id();
+    match api.get_account_data(&from_account_id).unwrap() {
+        Some(alice) => println!("[+] Alice's Free Balance is is {}\n", alice.free),
+        None => println!("[+] Alice's Free Balance is is 0\n"),
+    }
+
+    let to = AccountKeyring::Bob.to_account_id();
+
+    match api.get_account_data(&to).unwrap() {
+        Some(bob) => println!("[+] Bob's Free Balance is is {}\n", bob.free),
+        None => println!("[+] Bob's Free Balance is is 0\n"),
+    }
+    // generate extrinsic
+    let amount = 1000000000000000000;
+    let xt = api.balance_transfer(MultiAddress::Id(to.clone()), amount);
+
+    println!(
+        "Sending an extrinsic from Alice (Key = {}),\n\nto Bob (Key = {})\n",
+        from.public(),
+        to
+    );
+    println!("[+] Composed extrinsic: {:?}\n", xt);
+
+    // send and watch extrinsic until finalized
+    let tx_hash = api
+        .send_extrinsic(xt.hex_encode(), XtStatus::InBlock)
+        .unwrap();
+    println!("[+] Transaction got included. Hash: {:?}\n", tx_hash);
+
+    let (events_in, events_out) = channel();
+    api.subscribe_events(events_in).unwrap();
+    let args: ApiResult<TransferEventArgs> =
+        api.wait_for_event("Balances", "Transfer", None, &events_out);
+    match args {
+        Ok(transfer_event) => {
+            println!("Error : Transfer event received!!!\n");
```

Comment:
Is the error here correct?

Reply:
It should not get there, so it is in error if this happens

Reply:
Ah, I see. We want the error here. :) This was not clear to me, maybe be more clear about that?

Reply:
If it is an example, the error makes no sense. If it is a test, then we should improve the info message. For now I remove the Error in the message and we can update it if the examples shall test something. See comment from haerdib https://github.com/scs/substrate-api-client/pull/153#discussion_r695442582